test/test-httpd.js

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*eslint-disable*/
+
+const port = 8099;
+const file = require("sdk/io/file");
+const { pathFor } = require("sdk/system");
+const { Loader } = require("sdk/test/loader");
+const options = require("sdk/test/options");
+
+const loader = Loader(module);
+const httpd = loader.require("./lib/httpd");
+if (options.parseable || options.verbose)
+  loader.sandbox("./lib/httpd").DEBUG = true;
+
+exports.testBasicHTTPServer = function(assert, done) {
+  // Use the profile directory for the temporary file as that will be deleted
+  // when tests are complete
+  let basePath = pathFor("ProfD");
+  let filePath = file.join(basePath, 'test-httpd.txt');
+  let content = "This is the HTTPD test file.\n";
+  let fileStream = file.open(filePath, 'w');
+  fileStream.write(content);
```

Comment:
More seriously, I'm confused about why you write the file here, but then you don't actually read the file "onComplete" below (instead, `response.text` just get's compared to `content`). Is there voodoo magic happening that the httpd server is reading that file from disk when serving?